mds_checkout_fields.php

```diff
@@ -61,7 +61,51 @@ function mds_custom_checkout_field_update_order_meta( $order_id ) {
 		add_action( 'woocommerce_checkout_update_order_meta', 'mds_custom_checkout_field_update_order_meta' );
 	}
 
-	if ( ! function_exists( 'generate_towns' ) ) {
+    if ( ! function_exists( 'mds_custom_override_checkout_fields' ) ) {
+        /**
+         * Override the Billing and Shipping fields in Checkout.
+         *
+         * @param $address_fields
+         *
+         * @return array
+         */
+        function mds_custom_override_checkout_fields( $address_fields ) {
+            $mdsCheckoutFields = new \MdsSupportingClasses\MdsCheckoutFields( $address_fields );
+
+            $address_fields['billing']  = $mdsCheckoutFields->getCheckoutFields( 'billing' );
+            $address_fields['shipping'] = $mdsCheckoutFields->getCheckoutFields( 'shipping' );
+
+            $only_virtual = true;
+
+            foreach( WC()->cart->get_cart() as $cart_item_key => $cart_item ) {
+                // Check if there are non-virtual products
+                if ( ! $cart_item['data']->is_virtual() ) $only_virtual = false;
```

Comment:
Please stick to the wordpress code style.
See https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#brace-style

You should be able to set up your IDE to auto-format according to the wordpress standard.